tests/test_s3_excel.py

```diff
@@ -17,3 +18,19 @@ def test_excel(path, ext, use_threads):
     wr.s3.to_excel(df, file_path, use_threads=use_threads, index=False)
     df2 = wr.s3.read_excel(file_path, use_threads=use_threads)
     assert df.equals(df2)
+
+
+def test_read_xlsx_versioned(path) -> None:
+    df1 = pd.DataFrame({"c0": [0, 1, 2], "c1": [3, 4, 5]})
+    df2 = pd.DataFrame({"c0": [3, 4, 5], "c1": [6, 7, 8]})
+    path_file = f"{path}0.xlsx"
+    bucket, key = wr._utils.parse_path(path_file)
+    wr.s3.to_excel(df=df1, path=path_file, index=False)
+    wr.s3.to_excel(df=df2, path=path_file, index=False)
+    object_versions = [
+        v["VersionId"] for v in boto3.client("s3").list_object_versions(Bucket=bucket, Prefix=key)["Versions"]
+    ]
+    assert len(object_versions) == 2
+    for object_version, df in zip(object_versions, [df2, df1]):
+        df_temp = wr.s3.read_excel(path_file, version_id=object_version)
+        assert df_temp.equals(df)
```

Comment:
How about:
```
def test_read_xlsx_versioned(path) -> None:
    path_file = f"{path}0.xlsx"
    dfs = [pd.DataFrame({"c0": [0, 1, 2], "c1": [3, 4, 5]}), pd.DataFrame({"c0": [3, 4, 5], "c1": [6, 7, 8]})]

    for df in dfs:
        wr.s3.to_excel(df=df, path=path_file, index=False)
        version_id = wr.s3.describe_objects(path=path_file)[path_file]["VersionId"]
        df_temp = wr.s3.read_excel(path_file, version_id=version_id)
        assert df_temp.equals(df)
        assert version_id == wr.s3.describe_objects(path=path_file, version_id=version_id)[path_file]["VersionId"]
```
This way we also test `wr.s3.describe_objects` for more coverage and avoid using boto3 altogether